src/dmd/dinterpret.d

```diff
@@ -2163,37 +2159,39 @@ public:
             return;
         }
 
-        if (goal == ctfeNeedLvalue)
+        if (goal == CTFEGoal.LValue)
         {
-            VarDeclaration v = e.var.isVarDeclaration();
-            if (v && !v.isDataseg() && !v.isCTFE() && !istate)
-            {
-                e.error("variable `%s` cannot be read at compile time", v.toChars());
-                result = CTFEExp.cantexp;
-                return;
-            }
-            if (v && !hasValue(v))
+            if (auto v = e.var.isVarDeclaration())
             {
-                if (!v.isCTFE() && v.isDataseg())
-                    e.error("static variable `%s` cannot be read at compile time", v.toChars());
-                else // CTFE initiated from inside a function
+                if (!v.isDataseg() && !v.isCTFE() && !istate)
+                {
                     e.error("variable `%s` cannot be read at compile time", v.toChars());
-                result = CTFEExp.cantexp;
-                return;
-            }
-
-            if (v && (v.storage_class & (STC.out_ | STC.ref_)) && hasValue(v))
-            {
-                // Strip off the nest of ref variables
-                Expression ev = getValue(v);
-                if (ev.op == TOK.variable ||
-                    ev.op == TOK.index ||
-                    ev.op == TOK.slice ||
-                    ev.op == TOK.dotVariable)
+                    result = CTFEExp.cantexp;
+                    return;
+                }
+                if (!hasValue(v))
                 {
-                    result = interpret(pue, ev, istate, goal);
+                    if (!v.isCTFE() && v.isDataseg())
+                        e.error("static variable `%s` cannot be read at compile time", v.toChars());
+                    else // CTFE initiated from inside a function
+                        e.error("variable `%s` cannot be read at compile time", v.toChars());
+                    result = CTFEExp.cantexp;
                     return;
                 }
+
+                if (v.storage_class & (STC.out_ | STC.ref_))
```

Comment:
Please don't mix functional changes with refactoring.

Reply:
stoped the merge process until this review comment is resolved.

Reply:
It is not a functional change. If you look at the commit in isolation, it explains that this condition is redundant.

Reply:
fair enough